File: backend/4sae7-hot-stuff-dev-hot-stuff-dev/ReservationMicroService/src/main/java/com/example/reservationmicroservice/Entity/ReservationRestaurant.java
package com.example.reservationmicroservice.Entity;

import lombok.*;
import lombok.experimental.FieldDefaults;

import javax.persistence.*;
import java.io.Serializable;
import java.sql.Date;
import java.util.List;

@Entity
@AllArgsConstructor
@NoArgsConstructor
@Getter
@Setter
@EqualsAndHashCode
@ToString
@FieldDefaults(level = AccessLevel.PRIVATE)
public class ReservationRestaurant implements Serializable {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)

    Long idR ;
    Date dateStart;
    Date dateEnd;
    @Enumerated(EnumType.STRING)
    Pension pension ;
    @ElementCollection(targetClass = Allergy.class)
    @JoinTable(name = "tblAllergy", joinColumns = @JoinColumn(name = "idR"))
    @Column(name = "allergy", nullable = false)
    @Enumerated(EnumType.STRING)
    List<Allergy> allergies ;
    Integer nbrPerson;
    String nameGrp ;
    Boolean approved = false;
    Boolean expired = false;
    String username;
}

File: frontend/template_angular/src/app/components/data/SingleReservationFoyer/ReservationRestaurant.ts
export class ReservationRestaurant{
     idR !:number;
     dateStart!:Date;
     dateEnd!:Date;
     nbrPerson!: number;
     nameGrp !:String;
     approved!: Boolean;
     expired!: Boolean;
     username!: String;

}
export enum Pension {
    Demi_pension ='Demi_pension', 
    pension_complete='pension_complete',
}
export enum Allergy {
    Oeuf='Oeuf',
    Lait='Lait',
    Moutarde='Moutarde',
    Arachide='Arachide',
    Mollusques_et_crustacés='Mollusques_et_crustacés',
    Poissons='Poissons',
    Graines_de_sésame='Graines_de_sésame',
    Soja='Soja',
    Sulfites='Sulfites' ,
    Noix='Noix',
    Blé_et_triticale='Blé_et_triticale',

}